packages/jsii/lib/compiler.ts

```diff
@@ -246,6 +246,7 @@ export class Compiler implements Emitter {
       diagnostics.push(...assmEmit.diagnostics);
     } catch (e) {
       LOG.error(`Error during type model analysis: ${e}\n${e.stack}`);
+      hasErrors = true;
```

Comment:
This previously caused exceptions thrown from the assembler to be "swallowed" (after logging to `STDERR`) without necessarily causing the process to exit with non-0 (`hasErrors` could be `false` when this `catch` block triggers!).